cmd/entrypoint/main.go

```diff
@@ -41,8 +43,20 @@ var (
 )
 
 func main() {
+	// Add credential flags originally used in creds-init.
+	gitcreds.AddFlags(flag.CommandLine)
+	dockercreds.AddFlags(flag.CommandLine)
+
 	flag.Parse()
 
+	// Copy creds-init credentials from secret volume mounts to /tekton/creds
+	builders := []credentials.Builder{dockercreds.NewBuilder(), gitcreds.NewBuilder()}
+	for _, c := range builders {
+		if err := c.Write("/tekton/creds"); err != nil {
```

Comment:
Why do the credentials need to be written first to /tekton/creds and then copied to $HOME?  Is there a reason they can't be written directly to $HOME?

Reply:
@GregDritschler kinda yes. `/tektoncd/creds` is the "safe" path, aka it will always be there, it is under a reserved folder, … `HOME` might not be defined in the image. Depending on the runtime, the image, … it might point to different things, it might not have the correct rights, … 

/cc @sbwsg

Reply:
Totally right @vdemeester.  We have a variable expansion, `$(credentials.path)` that has previously expanded to either `/tekton/home` (when feature flag disabled - consistent $HOME) or `/tekton/creds` (when feature flag enabled - unknowable $HOME).  After this change it always resolves to `/tekton/creds` to ensure anyone using that variable still gets a valid value that points to the credentials.

This makes me wonder, though, whether we could mount all the credentials using a single projected volume into `/tekton/creds` without the extra copy steps.  I will investigate that in a follow-up after this PR.

Reply:
And I'm going to add a comment explaining this to this file.